product_supplierinfo_for_customer_sale/views/sale_view.xml

```diff
@@ -22,4 +22,79 @@
             </xpath>
         </field>
     </record>
+
+    <record id="product_customerinfo_tree_view" model="ir.ui.view">
+        <field name="name">product.customerinfo.tree.view</field>
+        <field name="model">product.customerinfo</field>
+        <field
+            name="inherit_id"
+            ref="product_supplierinfo_for_customer.product_customerinfo_tree_view"
+        />
+        <field name="arch" type="xml">
+            <xpath expr="//field[@name='name']" position="before">
+                <field name="product_tmpl_id" />
+            </xpath>
+        </field>
+    </record>
+
+    <record id="product_customerinfo_form_view" model="ir.ui.view">
+        <field name="name">product.customerinfo.form.view</field>
+        <field name="model">product.customerinfo</field>
+        <field
+            name="inherit_id"
+            ref="product_supplierinfo_for_customer.product_customerinfo_form_view"
+        />
+        <field name="arch" type="xml">
+            <xpath expr="//field[@name='product_id']" position="before">
+                <field name="product_tmpl_id" required="1" />
+            </xpath>
+        </field>
+    </record>
```

Comment:
**issue**: both of these add the field `product_tmpl_id` to every `customerinfo` view, but we don't want these fields, e.g., in the tree view embedded in the product

![image](https://github.com/OCA/sale-workflow/assets/78726989/9f28acc9-5076-46bf-8c4d-4f53efe9a36e)


A solution could be to mark these as `primary=True` and a priority of `99`. That way, these views won't be the default views. Then in the `act_window` below, specify that you want to use these views in particular.